openedx/core/djangoapps/user_authn/views/register.py

```diff
@@ -536,6 +536,7 @@ def post(self, request):
         redirect_url = get_redirect_url_with_host(root_url, redirect_to)
         response = self._create_response(request, {}, status_code=200, redirect_url=redirect_url)
         set_logged_in_cookies(request, response, user)
+        response.set_cookie(settings.EDX_SHOW_ACTIVATE_CTA, True) # setting the cookie for CTA dialogue
```

Comment:
We need to explicitly check here if the `user.is_active` is false then set this cookie because in some cases we don't send the account activation emails e.g. third-party auth etc. For more context check https://github.com/edx/edx-platform/blob/c8fcb22869b4d1ae2aebd9aabf4ce8a7558bf390/openedx/core/djangoapps/user_authn/views/register.py#L369-L423